modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -94,22 +94,17 @@ private long parseMillis(String date) {
                 format = format.substring(1);
             }
 
-            boolean isUtc = ZoneOffset.UTC.equals(zoneId);
-
             DateFormatter dateFormatter = DateFormatter.forPattern(format)
                 .withLocale(locale);
-            // if UTC zone is set here, the time zone specified in the format will be ignored, leading to wrong dates
-            if (isUtc == false) {
-                dateFormatter = dateFormatter.withZone(zoneId);
-            }
+
             final DateFormatter formatter = dateFormatter;
             return text -> {
                 TemporalAccessor accessor = formatter.parse(text);
                 // if there is no year nor year-of-era, we fall back to the current one and
                 // fill the rest of the date up with the parsed date
                 if (accessor.isSupported(ChronoField.YEAR) == false
                     && accessor.isSupported(ChronoField.YEAR_OF_ERA) == false
-                    && accessor.isSupported(WeekFields.of(locale).weekOfWeekBasedYear()) == false) {
+                    && accessor.isSupported(WeekFields.of(locale).weekBasedYear()) == false) {
```

Comment:
Is this change unrelated to the timezone issue?

Reply:
not related to the timezone issue, but also mentioned by a user https://github.com/elastic/elasticsearch/issues/63458#issuecomment-706503574

I realised I have note added a test case for it

Reply:
I think it would be better to isolate such a separate fix, so that the commit message is associated with what is being fixed, rather than a seemingly unrelated change.

Reply:
this is a good idea, better to isolate this change. 
A separate PR https://github.com/elastic/elasticsearch/pull/65717